x/govshuttle/keeper/keeper.go

```diff
@@ -3,36 +3,40 @@ package keeper
 import (
 	"fmt"
 
-	"github.com/cosmos/cosmos-sdk/store/prefix"
+	"cosmossdk.io/core/store"
+	"cosmossdk.io/store/prefix"
 	"github.com/ethereum/go-ethereum/common"
-	"github.com/tendermint/tendermint/libs/log"
 
+	"cosmossdk.io/log"
 	"github.com/Canto-Network/Canto/v7/x/govshuttle/types"
 	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/runtime"
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	govkeeper "github.com/cosmos/cosmos-sdk/x/gov/keeper"
 	paramtypes "github.com/cosmos/cosmos-sdk/x/params/types"
 )
 
 type (
 	Keeper struct {
-		storeKey   sdk.StoreKey
-		cdc        codec.BinaryCodec
-		paramstore paramtypes.Subspace
+		storeService store.KVStoreService
+		cdc          codec.BinaryCodec
+		paramstore   paramtypes.Subspace
 
 		accKeeper   types.AccountKeeper
 		erc20Keeper types.ERC20Keeper
-		govKeeper   types.GovKeeper
+		govKeeper   *govkeeper.Keeper
+		authority   string
```

Comment:
It appears that the assignment of `authority` in `NewKeeper` is missing, which might cause `MsgUpdateParams` to not function correctly.

Reply:
added `authority` to csr, inflation, onborading modules

Reply:
'ditto
> In the case of the `govshuttle` module, the assignment of the `authority` to `NewKeeper()` does not seem to be fixed yet. Since these actions related to authority are major changes occurring in this PR, it would be good to add integration and unit test cases for regression testing so that they can be caught in the workflow when not operating

Reply:
ditto
> The reason i didn't add authority and MsgUpdateParams to govshuttle is that the govshuttle module also has [Params](https://github.com/b-harvest/Canto/blob/dudong2/feat/canto-main-cosmos-sdk%40v0.50.x/proto/canto/govshuttle/v1/govshuttle.proto#L10), but they are empty. I skipped them because I don't need them right away.

Reply:
I See, then `GetAuthority()`, `UpdateParams()`, and `MsgUpdateParams` don't seem to be necessary for the govshuttle module, so why don't you erase them? Rather than having unnecessary authorities and msg in advance, it would be better to add them in terms of readability and error prevention if parameters occur in the future